pkg/util/ip_reservation.go

```diff
@@ -27,30 +27,35 @@ const (
 	reservationBlockSize   = "/29"
 )
 
-// IPAllocator struct maintains the current list of reserved /29 IPs pending a write operation in the cloud service. An IP Range is freed up from this list when it is available at the cloud provider end
-// As the create instance operation is in parallel, it is possible that an IP range was reserved by the service instance but is not updated in the cloud instance. The reserved list stores such IP ranges
+// IPAllocator struct consists of shared resources that are used to keep track of the /29 IPRanges currently reserved by service instances
 type IPAllocator struct {
-	hold      map[string]bool
+	// hold set maintains the set of  IP ranges that have been reserved by the service instances but pending reservation in the cloud instances
+	// The key is a IP range currently reserved by a service instance e.g(192.168.92.0/29). Value is a bool to implement map as a set
+	hold map[string]bool
+
+	// holdMutex is used to synchronize access to the hold set to prevent data races
 	holdMutex sync.Mutex
 }
 
 // NewIPAllocator is the constructor to initialize the IPAllocator object
-// Argument reserved map[string]bool is a set of currently reserved IPs for which the CreateInstance operation is not completed
+// Argument hold map[string]bool is a set of  IP ranges currently reserved by service instances but pending reservation in the cloud instances
 func NewIPAllocator(hold map[string]bool) *IPAllocator {
```

Comment:
Does hold need to be initialized by the caller, or can it be initialized inside the constructor?

Reply:
Why? right now I see the caller is just initializing it to empty

Reply:
Just using it as a parameterized constructor here where the caller has the flexibility to initialize it to any value rather than enforcing a default